src/test/kotlin/com/github/com/catppuccin/jetbrains_icons/providers/JavaIconProviderTest.kt

```diff
@@ -29,4 +46,312 @@ class JavaIconProviderTest : LightJavaCodeInsightFixtureTestCase5() {
+  fun getIcon_staticClass_hasStaticMark() {
+    fixture.addFileToProject(
+      "StaticClass.java",
+      """
+      public class OuterClass {
+        static class StaticClass { }
+      }
+      """
+        .trimIndent(),
+    )
+
+    val icon = runInEdtAndGet { provider.getIcon(fixture.findClass("OuterClass.StaticClass"), 1) }
+
+    // The icon should be a LayeredIcon with 2 layers - base icon and static mark
+    assertTrue(icon is LayeredIcon)
+    assertEquals(2, (icon as LayeredIcon).allLayers.size)
+    assertEquals(AllIcons.Nodes.StaticMark, icon.allLayers[1])
+  }
+
+  @Test
+  @DisplayName("Test that a non-static class icon does not include a static mark")
+  fun getIcon_nonStaticClass_hasNoStaticMark() {
+    fixture.addFileToProject(
+      "RegularClass.java",
+      """
+      public class RegularClass { }
+      """
+        .trimIndent(),
+    )
+
+    val icon = runInEdtAndGet { provider.getIcon(fixture.findClass("RegularClass"), 1) }
+
+    // For non-static classes, the icon should not be layered
+    assertFalse(icon is LayeredIcon)
+  }
+
+  @Test
+  @DisplayName("Test that requesting an icon for a non-PsiClass element returns null")
+  fun getIcon_notPsiClass_returnsNull() {
+    fixture.addFileToProject(
+      "SomeClass.java",
+      """
+      class SomeClass {
+        private int someField;
+      }
+      """
+        .trimIndent(),
+    )
+
+    val icon = runInEdtAndGet { provider.getIcon(fixture.findClass("SomeClass").fields[0], 1) }
+
+    assertNull(icon)
+  }
+
+  @Test
+  @DisplayName("Test that requesting an icon for a non-Java file returns null")
+  fun getIcon_nonJavaFile_returnsNull() {
+    fixture.addFileToProject(
+      "SomeClass.kt",
+      """
+      class SomeClass { }
+      """
+        .trimIndent(),
+    )
+
+    val icon = runInEdtAndGet { provider.getIcon(fixture.findClass("SomeClass"), 1) }
+
+    assertNull(icon)
+  }
+
+  @Test
+  @DisplayName("Test that when Java support is disabled, a generic Java icon is returned")
+  fun getIcon_javaDisabled_returnsJavaIcon() {
+    fixture.addFileToProject(
+      "SomeClass.java",
+      """
+      interface SomeClass { }
+      """
+        .trimIndent(),
+    )
+
+    // Temporarily disable Java support
+    val originalSetting = PluginSettingsState.instance.javaSupport
+    PluginSettingsState.instance.javaSupport = false
+
+    val icon = runInEdtAndGet { provider.getIcon(fixture.findClass("SomeClass"), 1) }
+
+    assertEquals(icons.java, icon)
+
+    // Restore original setting
+    PluginSettingsState.instance.javaSupport = originalSetting
+  }
+
+  @Test
+  @DisplayName("Test visibility icon for a public class")
+  fun getVisibilityIconForTesting_publicClass_returnsPublicIcon() {
+    fixture.addFileToProject(
+      "PublicClass.java",
+      """
+    public class PublicClass { }
+    """
+        .trimIndent(),
+    )
+
+    val icon = runInEdtAndGet { provider.getVisibilityIcon(fixture.findClass("PublicClass")) }
```

Comment:
You only need to test the public interface. Testing internal methods makes tests and code brittle